Userland/Libraries/LibJS/Heap/Heap.cpp

```diff
@@ -191,9 +191,17 @@ class MarkingVisitor final : public Cell::Visitor {
 void Heap::mark_live_cells(const HashTable<Cell*>& roots)
 {
     dbgln_if(HEAP_DEBUG, "mark_live_cells:");
+
+    for (auto& inverse_root : m_inverse_roots)
+        inverse_root->set_marked(false);
+
```

Comment:
Do we want to do this first pass? 
We technically don't need it but it could leave non referenced cells if an inverse root is also a normal root.

Reply:
I'm not sure what this pass is supposed to achieve? if the inverse root is a root itself all of its descendants will still be marked, meaning this is essentially a no-op, what you can try and do is remove them from the roots list here, but i don't think that really matters, it just means its descendants will be collected in the next GC pass instead of the current one, which is allowed by the spec.

Reply:
You're right I misread the following loop in that case let's just skip it.